gadget/install/encrypt.go

```diff
@@ -35,23 +35,33 @@ var (
 	secbootAddRecoveryKey        = secboot.AddRecoveryKey
 )
 
-// encryptedDevice represents a LUKS-backed encrypted block device.
-type encryptedDevice struct {
+// encryptedDeviceCryptsetup represents a encrypted block device.
+type encryptedDevice interface {
+	Node() string
+	AddRecoveryKey(key secboot.EncryptionKey, rkey secboot.RecoveryKey) error
+	Close() error
+}
+
+// encryptedDeviceLUKS represents a LUKS-backed encrypted block device.
+type encryptedDeviceLUKS struct {
 	parent *gadget.OnDiskStructure
 	name   string
-	Node   string
+	node   string
 }
 
+// sanity
+var _ = encryptedDevice(&encryptedDeviceLUKS{})
```

Comment:
Is the compiler smart enough to remove this?

In any case, you already have real code that uses `encryptedDeviceLUKS` as an `encryptedDevice`, so maybe this can be removed?

(in case the interface becomes unused, a static analyzer would be able to detect it if you remove this line here; but if you leave it, I'm not sure it could)

Reply:
Given that it is used already I guess it could be removed, yes.

Reply:
I like to keep these sorts of lines here even if they are technically redundant since they offer a convenient place where we see that a given struct is meant to implement a given interface and it also provides a nice compiler error when the struct doesn't implement the interface anymore and at least in my editor makes it glaringly obvious what changed as I can see this line highlighted as causing a compiler error

Reply:
but also yes the compiler is smart enough to remove this

Reply:
OK then, fine with me :-)